app/services/api/to_user_sessions_hash2.rb

```diff
@@ -0,0 +1,73 @@
+class Api::ToUserSessionsHash2
+  def initialize(form:, user:)
+    @form = form
+    @user = user
+  end
+
+  def call
+    return Failure.new(form.errors) if form.invalid?
+
+    delete_sessions(data.select(&:deleted))
+
+    Success.new(
+      {
+        upload: new_in_mobile_app,
+        download: new_in_db + outdated,
+        deleted: deleted
+      }
+    )
+  end
+
+  private
+
+  attr_reader :form, :user
+
+  def data
+    form.to_h.data
+  end
+
+  def delete_sessions(sessions)
+    user.sessions.where(uuid: sessions.map(&:uuid)).destroy_all
+  end
+
+  def deleted
+    DeletedSession.where(user: user, uuid: data.map(&:uuid)).map(&:uuid)
+  end
+
+  def new_in_mobile_app
+    uuids_present_in_mobile_app - uuids_present_in_db - deleted
+  end
+
+  def uuids_present_in_mobile_app
+    uuids_present_in_mobile_app ||= present_in_mobile_app.map(&:uuid)
```

Comment:
Why the `=`?

Reply:
Not really sure what this does to be honest :D

Reply:
Actually, I forgot about @ at the beginning of `@uuids_present_in_mobile_app`
The idea is that the calculation on the right only happens once, on the first call to the method. The next time the value is reused. Of course, it doesn't work if it's a local variable 🤦‍♂